Core: Switch tests to Junit5 in metrics,puffin,view,rest pakages

skytin1004 opened the pull request:
I switched the files in [metrics,puffin] pakages to JUnit5.
I also switched the files in [requests, responses] in rest pakage to JUnit5


I modified the view package and committed it(commit name: Core: Switch tests to Junit5 in view), but since this part already exists in another PR I posted (Core: Switch tests to Junit5 in catalog,encryption,inmemory,io,view), I reverted it back.

nastra commented on core/src/test/java/org/apache/iceberg/rest/responses/TestCreateNamespaceResponse.java:
```diff
@@ -165,8 +163,12 @@ public CreateNamespaceResponse createExampleInstance() {
-    Assert.assertEquals("Namespaces should be equal", actual.namespace(), expected.namespace());
-    Assert.assertEquals("Properties should be equal", actual.properties(), expected.properties());
+    Assertions.assertThat(actual.namespace())
+        .as("Namespaces should be equal")
```
I don't think the `.as()` provides any additional help during debugging, so we can omit it here

nastra commented on core/src/test/java/org/apache/iceberg/rest/responses/TestGetNamespaceResponse.java:
```diff
@@ -67,71 +66,64 @@ public void testCanDeserializeWithoutDefaultValues() throws JsonProcessingExcept
-        () -> deserialize(jsonPropertiesHasWrongType));
+    Assertions.assertThatThrownBy(() -> deserialize(jsonPropertiesHasWrongType))
+        .as("A JSON response with the wrong type for a field should fail to deserialize")
+        .isInstanceOf(JsonProcessingException.class);
```
hasMessage(..) here and in other files is missing

nastra commented on core/src/test/java/org/apache/iceberg/rest/responses/TestUpdateNamespacePropertiesResponse.java:
```diff
@@ -145,92 +143,98 @@ public void testDeserializeInvalidResponse() {
-        () -> deserialize(null));
+    Assertions.assertThatThrownBy(() -> deserialize(null))
+        .as("A null JSON response body should fail to deserialize")
+        .isInstanceOf(IllegalArgumentException.class);
```
hasMessage()

skytin1004 commented:
Thank you for the detailed review, @nastra.
Based on your review, I went back through all the code in this PR and deleted the `.as(..) `and added a  `.hasMessage(..)` method in `assertThatThrownBy` . 

In addition, I deleted the unnecessary as(..) in the part you suggested ("Namespaces should be equal" and "Properties should be equal") and 
   `Assert.assertFalse(
        "The merged properties map should omit keys with null values", merged.containsValue(null))` I changed this to code that uses AssertJ using the method you suggested.

nastra reviewed (approved):
LGTM, thanks @skytin1004
